crates/nassun/src/client.rs

```diff
@@ -57,6 +58,15 @@ impl NassunOpts {
         self
     }
 
+    /// Set the credential-config for this instance. The config has the form (registry,key,value)
+    /// where registry is the host of the registry, key is the key of a single entry in the respective
+    /// config JSON object and value is the corresponding value. A single registry may have multiple entries
+    /// in this vector (e.g. "username" & "password"). The format will be handled by by OroClient.
+    pub fn credentials(mut self, credentials: Vec<(String, String, String)>) -> Self {
+        self.credentials = credentials;
+        self
+    }
```

Comment:
you want this option in node-maintainer, too, and make sure it gets passed down directly to nassun_opts.

Reply:
I've added a `fn credentials(...)` to `NodeMaintainerOptions` and just hand those over to the contained `nassun_opts` there.